### PR TITLE
HPCC-14668 Fix destructors that can throw exceptions

### PR DESCRIPTION
**Pull request by richardkchapman**
Few good fixes available.

Signed-off-by: Richard Chapman rchapman@hpccsystems.com

### CODE REVIEW

**Comment by richardkchapman** [comment]
@ghalliday @jakesmith First stab at this for comment...

File: dali/base/dasds.cpp [hunk 1]
```diff
@@ -5236,7 +5236,7 @@ class CStoreHelper : public CInterface, implements IStoreHelper
                 MilliSleep(100);
             }
         }
-        ~CheckDeltaBlock()
+        ~CheckDeltaBlock() noexcept(false)
```
Comment (jakesmith) [inline dali/base/dasds.cpp]: Should we not instead remove the throw in the destructor and catch/keep the exception.
Then ensure a member function is called to throw it?
(There are a couple of instances of CheckDeltaBlock, would probably need to ensure that the scope they're, is in a try/catch block and call after)

File: system/jlib/jiface.hpp [hunk 2]
```diff
@@ -44,18 +44,16 @@ void jlib_decl raiseAssertCore(const char *assertion, const char *file, unsigned
  #define assertex(p)
  #define assert(p)
 #endif
-
 #if defined(_DEBUG)||defined(_TESTING)
 #define verifyex(p) (likely(p) ? ((void) 0) : (( (void) raiseAssertException(#p, __FILE__, __LINE__))))
 #define verify(p) (likely(p) ? ((void) 0) : (( (void) raiseAssertCore(#p, __FILE__, __LINE__))))
 #else
-#define verifyex(p) ((void) (p))
+ #define verifyex(p) ((void) (p))
```
Comment (jakesmith) [inline system/jlib/jiface.hpp]: trivial: accidental space insertion I assume

File: thorlcr/slave/slave.cpp [hunk 3]
```diff
@@ -66,6 +68,10 @@ ProcessSlaveActivity::~ProcessSlaveActivity()
     ActPrintLog("AFTER ProcessSlaveActivity : joining process thread");
 }
 
+ProcessSlaveActivity::~ProcessSlaveActivity()
+{
+}
```
Comment (jakesmith) [inline thorlcr/slave/slave.cpp]: I'd have probably deleted it, now empty

**Comment by jakesmith** [comment]
@richardkchapman - looks good, one question and couple of minors

File: dali/base/dasds.cpp (hunk 1 above)
Comment (richardkchapman) [inline dali/base/dasds.cpp]: The whole purpose of CheckDeltaBlock (as far as I can tell) is to throw the exception if the check fails, and to ensure that the check is done.

File: thorlcr/slave/slave.cpp (hunk 3 above)
Comment (richardkchapman) [inline thorlcr/slave/slave.cpp]: Yes, not sure why I didn't

File: system/jlib/jiface.hpp (hunk 2 above)
Comment (richardkchapman) [inline system/jlib/jiface.hpp]: Well, moved from a few lines up where the space was right. I'll make the leading spaces consistent.

**Pull request merged**